### ISSUE TITLE
Add physical CDROM interface support for macOS

### ISSUE THREAD

**Issue by Grounded0** [dosbox-staging/dosbox-staging#3602]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

macOS (Intel)

### Is your feature request related to a problem? Please describe.

Since Windows and Linux will have functioning physical CD-ROM interfaces soon i'm adding this as a functionality gap for macOS:

- #3591

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
Anyone interested in this? @kklobe perhaphs?

**Comment by Grounded0** [comment]
This is super niche feature for macOS as optical drives were killed off on Macs a long ago. I left this as a reminder that we have a functionality gap between the OSes which we usually don't do.

**Comment by johnnovak** [comment]
> This is niche feature for macOS as optical drives were killed off on Macs a long time ago. I left this as a reminder that we have a functionality gap between the OSes which we usually don't have.

Yeah, hence I asked the question, mostly for feature parity rather than anything else.

**Comment by Grounded0** [comment]
I was thinking why on earth you would want this but then I realised: copy protection.

If you mount copy protected disc images via DAEMON Tools you need access to the OS level optical disc support for its optical disc simulation to work. Since we only have to deal with LaserLock for DOS currently its a non-issue but could be helpful for people using DOSBox Staging with Windows 9x.

**Comment by Grounded0** [comment]
Removing https://github.com/dosbox-staging/dosbox-staging/labels/audio since this has nothing to do with audio. Our own disc image mounter already provides Red Book audio support.

**Comment by johnnovak** [comment]
> Dropping [ audio](https://github.com/dosbox-staging/dosbox-staging/labels/audio) Audio related issues or enhancements since this has nothing to do with audio. Our `IMGMOUNT` already provides Red Book audio support.

Not true. Red Book audio from physical CD-ROMs are mounted through the mixer now on Windows and Linux, so this is audio related as well. Restoring label.

**Comment by weirddan455** [comment]
>  Red Book audio from physical CD-ROMs is now routed through the mixer on Linux, and we should add this for Windows for feature parity.

This is done by the way since #3591

It's just Mac that is lacking physical CDROM support because that's OS specific code and I don't have a Mac.

**Comment by johnnovak** [comment]
> This is done by the way since [#3591](https://github.com/dosbox-staging/dosbox-staging/pull/3591)

I've ticked the windows item in the top description.

> It's just Mac that is lacking physical CDROM support because that's OS specific code and I don't have a Mac.

I've heard @FeralChild64 is big fan of CD-ROMs, so maybe he'll be interested in tinkering a bit with adding macOS support. I can contribute with testing, I still have two USB CD/DVD writers.

**Comment by FeralChild64** [comment]
It's on my personal TODO list, but this list is rather lengthy :)
